onnxruntime/core/providers/cuda/tensor/gather_elements.cc

```diff
@@ -118,6 +118,24 @@ ONNX_NAMESPACE::TensorProto_DataType GetElementType(size_t element_size) {
   }
 }
 
+// Get Element Type by DataTypeImpl
+ONNX_NAMESPACE::TensorProto_DataType GetElementType(const DataTypeImpl* dtype) {
```

Comment:
I believe there is an util for the same functionality. Let's try to find that one.